packages/next-auth/src/next/middleware.ts

```diff
@@ -101,19 +101,20 @@ async function handleMiddleware(
   options: NextAuthMiddlewareOptions | undefined,
   onSuccess?: (token: JWT | null) => Promise<NextMiddlewareResult>
 ) {
-  const { pathname, search, origin } = req.nextUrl
+  const { pathname, search, origin, basePath: nextJsBasePath } = req.nextUrl
 
-  const signInPage = options?.pages?.signIn ?? "/api/auth/signin"
-  const errorPage = options?.pages?.error ?? "/api/auth/error"
+  const fullPathname = nextJsBasePath + pathname
+  const signInPage = options?.pages?.signIn ?? `${nextJsBasePath}/api/auth/signin`
```

Comment:
Adding basePath into signIn and error page from start this function is bad idea in my opinion, because code for check urls for skip is getting complicated after this.

I advice to remove it from this code and add when construct resulted url via new URL when error happened or call redirect
@ThangHuuVu what do you think about this?

Reply:
I agree removing it here, but I have to add it at least in line 129 and 149 to have a proper redirection. We have to prepend the nextjs to the sign in and error page on redirect.

Reply:
I adapted accordingly